Introduce recipes (#1538)
saleor/saleor-docs: docs/recipes/click-and-collect.mdx

---
title: Click and collect
---

import CardGrid from "@site/components/CardGrid";
import CompactCard from "@site/components/CompactCard";

# Click and collect recipe 

Saleor's click and collect functionality enables customers to order products online and collect them from physical locations. This hybrid e-commerce model satisfies customers who want to minimize time spent in stores while still shopping from their favorite retailers. This report provides a comprehensive guide to configuring and implementing click and collect in your Saleor-powered e-commerce platform.

<div style={{display: 'flex', gap: '1rem', flexWrap: 'wrap', paddingBottom: '4rem'}}>
    <CompactCard
        name="Lush Case Study"
        description="Learn more how Lush is using this feature."
        link="https://saleor.io/case-study/lush"
    />
</div>

# Setting up collection points

Each physical location where customers can collect orders must be represented as a **Warehouse** in Saleor, even if you don't track inventory at that specific location. The Warehouse entity represents the physical address for pickup.

> By choosing allocation strategy, you can choose if different warehouses will able to deliver to this location, or customers will be able to order only the products available on site.

Reviewer: **suggestion**: I think we should more explicitly say something like:

> You should select "all warehouses" In the "Pickup" section of the Warehouse screen.

<div style={{display: 'flex', gap: '1rem', flexWrap: 'wrap', paddingBottom: '4rem'}}>
    <CompactCard
        name="Warehouses docs"
        description="Read more about warehouses configuration."
        link="/developer/stock/stock-allocation"
    />
</div>

# Storefront changes

> List of available collection points for given checkout can be found in `availableCollectionPoints` field. 

Reviewer: **suggestion**:

```suggestion
List of available collection points for given checkout can be found in `checkout.availableCollectionPoints` field. 
```

Ideally, we could also add an example query.

> To select it use `checkoutDeliveryMethodUpdate` mutation. Shipping address of the checkout will be updated to chosen collection point.

Reviewer: ```suggestion
To select it, use `checkoutDeliveryMethodUpdate` mutation. Shipping address of the checkout will be updated to chosen collection point.
```

<div style={{display: 'flex', gap: '1rem', flexWrap: 'wrap', paddingBottom: '4rem'}}>
    <CompactCard
        name="Saleor Storefront quickstart"
        description="Try example storefront build with NextJS."
        link="/quickstart/storefront"
    />

    <CompactCard
        name="Checkout shipping docs"
        description="Read more about shipping and checkout APIs."
        link="/developer/checkout/address"
    />
</div>

# Delivery status

When order is picked up, staff users can use Saleor Dashboard to update fulfillments status as delivered. 

> For integration with existing order management software create app which will automatically synchronize status with Saleor API.

Reviewer: **style**:

```suggestion
To integrate with existing order management software, create an app which will automatically synchronize status with Saleor API.
```
